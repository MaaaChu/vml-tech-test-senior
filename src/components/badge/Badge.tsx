import React from "react";
import classNames from "classnames";

import styles from "./Badge.module.css";

interface BadgeProps {
  children: React.ReactNode;
  type: "primary";
}

const Badge: React.FC<BadgeProps> = ({ children, type = "primary" }) => {
  return (
    <div className={classNames(styles.badge, styles[`badge--${type}`])}>
      <span className={styles["badge__text"]}>{children}</span>
    </div>
  );
};

export default Badge;
