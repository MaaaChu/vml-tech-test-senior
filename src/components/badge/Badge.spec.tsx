import React from "react";
import "@testing-library/jest-dom";
import { render, RenderResult } from "@testing-library/react";
import Badge from "./Badge";

describe("components:Badge", () => {
  let wrapper: RenderResult;
  let text: string;

  beforeEach(async () => {
    text = "test";
    wrapper = render(<Badge type="primary">{text}</Badge>);
  });

  it("should match snapshot", () => {
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it("should render badge text", () => {
    const linkElement = wrapper.getByText(/test/i);
    expect(linkElement).toBeInTheDocument();
  });
});
