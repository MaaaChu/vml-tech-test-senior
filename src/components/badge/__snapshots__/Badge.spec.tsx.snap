// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Badge should match snapshot 1`] = `
<div
  class="badge badge--primary"
>
  <span
    class="badge__text"
  >
    test
  </span>
</div>
`;
